### PR TITLE
[WIP] Clarify source comments and add details to cli option help output.

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- Improve help output by reorganizing and clarifying valid options and usage.

Resolves https://github.com/LLNL/RAJAPerf/issues/353

### CODE REVIEW

**Comment by rhornung67** [comment]
@MrBurmark please take a look at this when you have time and provide feedback. I'm going to look at improving error checking support.

**Comment by rhornung67** [comment]
@artv3 I tagged you on this since you may be interested.

**Review by artv3 — APPROVED** [review]

File: src/common/RunParams.cpp [hunk 1]
```diff
@@ -718,18 +729,18 @@ void RunParams::parseCommandLineOptions(int argc, char** argv)
...
 {
   str << "\nUsage: ./raja-perf.exe [options]\n";
+
   str << "Valid options are:\n";
 
-  str << "\t --help, -h (print options with descriptions)\n\n";
+  str << "\t --help, -h (print available command line options with descriptions)\n\n";
 
-  str << "\t --show-progress, -sp (print execution progress during run)\n\n";
+  str << "\t Print info about option arguments....\n"
```
Comment (MrBurmark) [inline src/common/RunParams.cpp:737]: I didn't understand what this meant the first few times I read it. Should it say something more like "Options to prints info ..."

Comment (rhornung67) [reply src/common/RunParams.cpp]: Good point. I changed it.

Comment (MrBurmark) [reply src/common/RunParams.cpp]: Can you add something like "Options to ..." to the other category intros?

Comment (rhornung67) [reply src/common/RunParams.cpp]: Done. If you're OK with this, I will move it out of draft stage so all CI checks will run

**Review by MrBurmark — APPROVED** [review]

**Comment by MrBurmark** [comment]
This is much better organized and explained.

**Pull request merged**